ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -185,10 +188,29 @@ public Void visitFunctionCall(final FunctionCall node, final TypeContext context
       final List<SqlArgument> argumentTypes = new ArrayList<>();
       final FunctionName functionName = node.getName();
       for (final Expression argExpr : node.getArguments()) {
-        process(argExpr, null);
-        argumentTypes.add(SqlArgument.of(
-            expressionTypeManager.getExpressionSqlType(argExpr)
-        ));
+        process(argExpr, context);
+        final SqlType newSqlType = expressionTypeManager.getExpressionSqlType(argExpr, context);
```

Comment:
```suggestion
        final SqlType resolvedArgType = expressionTypeManager.getExpressionSqlType(argExpr, context);
```
nit: "new" is a little confusing for me